akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -2322,6 +2345,10 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
       }
     }
 
+  def receiveMemberExiting(m: Member): Unit =
+    if (matchingRole(m) && m.address != selfAddress)
+      exitingNodes += m.uniqueAddress
```

Comment:
What about ongoing write aggregation, and a race where the write aggregation started, a node exits and the write consistency can no longer be fulfilled (but that should perhaps be ok)

Reply:
I don't think we need to make it too advanced and modify read/write aggregators that are already in flight.